# Assertion Message: At Least One Assertion is Required

Rule codename: _RuleAssertionMessage_
___

Each test case has to contain at least one assertion. The assertion message
should be meaningful and provide information about the expected and actual
values or behaviour.

Wrong:
```java
@Test
void checks() {
    final int seed = 127;
    final int actual = new Random(seed).nextInt();
    final int expected = -1190496726;
    if(actual != expected) {
        throw new AssertionError(
            String.format(
                "We expect that actual value generated by seed %s is equal to %s value",
                seed,
                expected
            )
        );
    }
}
```

Correct:
```java
@Test
void checks() {
    final int seed = 127;
    final int actual = new Random(seed).nextInt();
    final int expected = -1190496726;
    MatcherAssert.assertThat(
        String.format(
            "We expect that actual value generated by seed %s is equal to %s value",
            seed,
            expected
        ),
        actual,
        Matchers.is(expected)
    );
}
```

We support only Hamcrest and JUnit 5 assertions. 

Exceptions:
`@SuppressedWarnings("JTCOP.RuleAssertionMessage")`.

You can read more about that
rule [here](https://www.yegor256.com/2023/01/19/layout-of-tests.html#assertions).